components/molecules/m-product-gallery.vue

```diff
@@ -57,11 +55,19 @@ export default {
       };
     },
     variantImage () {
-      let variantImage = this.gallery.find(
-        imageObject =>
-          isEqual(imageObject.id, this.option) ||
-          (imageObject.id && imageObject.id.color === this.option.color)
-      );
+      let variantImage = this.gallery.find(image => {
+        let selectThis = true
+        for (const [key, value] of Object.entries(this.configuration)) {
+          if (
+            typeof image.id !== 'undefined' &&
+            typeof image.id[key] !== 'undefined' &&
+            image.id[key] !== value.id
+          ) {
+            selectThis = false
+          }
+        }
+        return selectThis
+      })
```

Comment:
I'm not sure but this logic will not find partially matched product configurations, i.e. it will not find black image if configuration will be size 36 and color black -- in case there is no dedicated image for this configuration  🤔

Reply:
You are right. I fixed this.